src/main/java/org/candlepin/model/ContentAccessCertificateCurator.java

```diff
@@ -218,4 +218,35 @@ public List<CertSerial> listCertSerials(Collection<String> consumerIds) {
         return serials;
     }
 
+    /**
+     * Deletes the content access certificates for the provided consumers and also revokes the certificate
+     * serials.
+     *
+     * @param consumerUuids
+     *  the UUIDs of {@link Consumer}s to delete content access certificates for
+     *
+     * @return the number of content access certificates that were deleted
+     */
+    public int deleteForConsumers(Collection<String> consumerUuids) {
+        if (consumerUuids == null || consumerUuids.isEmpty()) {
+            return 0;
+        }
+
+        String hql = """
+            SELECT cac.id, s.id
+            FROM Consumer c
+                JOIN c.contentAccessCert cac
+                JOIN cac.serial s
+            WHERE c.uuid IN (:uuids)
+            """;
+
+        Query query = this.getEntityManager()
+            .createQuery(hql);
+
+        List<Object[]> rows = query
```

Comment:
Missing partitioning on consumerUuids

Reply:
Good catch! Added partitioning on the consumerUuids